test/csuite/incr_backup/main.c

```diff
@@ -522,182 +381,60 @@ tables_free(TABLE_INFO *tinfo)
-    char buf[4096], copy_from[PATH_MAX], copy_to[PATH_MAX];
-    char *filename;
-    char granularity_unit;
-    const char *cons;
+    uint32_t granularity, granularity_kb;
+    int id, nfiles;
+    bool consolidate;
 
     nfiles = 0;
-
-    VERBOSE(2, "BASE BACKUP: %s\n", backup_home);
-    active_files_free(active);
-    active_files_init(active);
-    testutil_recreate_dir(backup_home);
-
-    testutil_check(conn->open_session(conn, NULL, NULL, &session));
-    tinfo->full_backup_number = tinfo->incr_backup_number++;
+    id = (int)tinfo->full_backup_number;
 
     /* Half of the runs with very low granularity to stress bitmaps */
     granularity = __wt_random(rand) % 20;
     if (__wt_random(rand) % 2 == 0) {
-        granularity_unit = 'K';
         granularity += 4;
+        granularity_kb = granularity;
     } else {
-        granularity_unit = 'M';
         granularity += 1;
+        granularity_kb = granularity * 1024;
     }
     if (__wt_random(rand) % 2 == 0)
-        cons = ",consolidate=true";
+        consolidate = true;
     else
-        cons = ",consolidate=false";
-    testutil_snprintf(buf, sizeof(buf),
-      "incremental=(granularity=%" PRIu32 "%c,enabled=true,%s,this_id=ID%" PRIu32 ")", granularity,
-      granularity_unit, cons, tinfo->full_backup_number);
-    VERBOSE(3, "open_cursor(session, \"backup:\", NULL, \"%s\", &cursor)\n", buf);
-    testutil_check(session->open_cursor(session, "backup:", NULL, buf, &cursor));
-
-    while ((ret = cursor->next(cursor)) == 0) {
-        nfiles++;
-        testutil_check(cursor->get_key(cursor, &filename));
-        active_files_add(active, filename);
-        testutil_snprintf(copy_from, sizeof(copy_from), "%s/%s", home, filename);
-        testutil_snprintf(copy_to, sizeof(copy_to), "%s/%s", backup_home, filename);
-        VERBOSE(3, " => copy %s %s\n", copy_from, copy_to);
-        testutil_copy(copy_from, copy_to);
-    }
-    testutil_assert(ret == WT_NOTFOUND);
-    testutil_check(cursor->close(cursor));
-    testutil_check(session->close(session, NULL));
-    active_files_sort(active);
+        consolidate = false;
+    /* Use the same ID for the directory name and configuration */
+    testutil_backup_create_full(conn, home, id, consolidate, granularity_kb, &nfiles);
     VERBOSE(2, " finished base backup: %d files\n", nfiles);
 }
 
-/*
- * reopen_file --
- *     Open a file if it isn't already open. The "memory" of the open file name is kept in the
- *     buffer passed in.
- */
-static void
-reopen_file(int *fdp, char *buf, size_t buflen, const char *filename, int oflag)
-{
-    /* Do we already have this file open? */
-    if (strcmp(buf, filename) == 0 && *fdp != -1)
-        return;
-    if (*fdp != -1)
-        testutil_check(close(*fdp));
-    *fdp = open(filename, oflag, 0666);
-    strncpy(buf, filename, buflen);
-    testutil_assert(*fdp >= 0);
-}
-
 /*
  * incr_backup --
  *     Perform an incremental backup into an existing backup directory.
  */
```

Comment:
Add a comment to identify the directories used such as the below.
`*   The directory containing the source is BACKUP_BASE.tinfo->full_backup_number.`
`*   The new incremental backup directory is BACKUP_BASE.tinfo->incr_backup_number.`